SO-3947: Make maximum file upload configurable

kovacstomi opened the pull request.

Commit:
SO-3947: add commons-fileupload to snowowl.core.rest dependencies

Commit:
SO-3947: remove java configured properties

Commit:
SO-3947: add new fields to read by ApiConfig bootstrap

Commit:
SO-3947: add class to store file upload configuration properties

Commit:
SO-3947: configure MultipartResolver with the new registered properties

apeteri commented on core/com.b2international.snowowl.core.rest/build.properties:
```diff
@@ -3,4 +3,5 @@ output.. = target/classes
                WEB-INF/,\
-               site/
+               site/,\
+               WEB-INF/lib/commons-fileupload-1.4.jar
```
`WEB-INF/` above includes this file, no need to change.

cmark replied:
Done!

cmark committed:
SO-3947 #resolve

Final changes to upload file size configuration.
Increase max allowed upload file size to `1gb`.
Configure it through the `api.http.maxFileSize` and
`api.http.maxRequestSize` configuration settings.

cmark reviewed (approved):
LGTM! Few changes made to the code regarding configuration and default upload sizes.

cmark committed:
SO-3947: add NotEmpty annotations to HttpConfig fields
